packages/back-end/src/controllers/features.ts

```diff
@@ -1211,6 +1332,43 @@ export async function postFeatureMoveRule(
     version: revision.version,
   });
 }
+export async function getDraftandReviewRevisions(
+  req: AuthRequest,
+  res: Response
+) {
+  const context = getContextFromReq(req);
+  const revisions = await getRevisionsByStatus(context, [
+    "draft",
+    "approved",
+    "changes-requested",
+    "pending-review",
+  ]);
+  const featuresAndRevisions: {
+    revision: FeatureRevisionInterface;
+    feature: FeatureInterface | null;
+  }[] = [];
+
+  for (const revision of revisions) {
+    if (revision) {
+      const feature = await getFeature(context, revision.featureId);
```

Comment:
Don't run Mongo queries inside of loops.  In this particular case, it might be better to move this filtering logic to the front-end instead.  The front-end already has all of the feature objects in memory, so it won't require any additional db lookups or API calls.